index.js

```diff
@@ -1,2 +1,37 @@
+/* eslint-disable func-names, no-extra-bind */
+
 'use strict';
-module.exports = (fn, name) => Object.defineProperty(fn, 'name', {value: name, configurable: true});
+module.exports = (fn, name) => {
+	Object.defineProperty(fn, 'name', {value: name, configurable: true});
+	const functionString = fn.toString();
+	if (
+		!(
+			!/^(?:async\s*)?(?:(\/\*[^]*?\*\/|\/\/[^]*?(?:\r\n|\r|\n))\s*)*(?:function)?\s*(?:(?:\/\*[^]*?\*\/|\/\/[^]*?(?:\r\n|\r|\n))\s*)*\*?.*?\s*(?:\/\*[^]*?\*\/|\/\/[^]*?(?:\r\n|\r|\n)\s*)*\(/.test(functionString) ||
+			functionString
+				.replace(/(async|function|\s*)/g, '')
+				.replace(/(\/\*[^]*?\*\/|\/\/[^]*?(?:\r\n|\r|\n))/g)
+				.trim()
+				.startsWith('(')
+		) &&
+		functionString.indexOf('class') !== 0 &&
+		(!/^(get|set)/.test(functionString) || !functionString.slice(fn.name.length).replace(/\s/g, '').replace(/(\/\*[^]*?\*\/|\/\/[^]*?(?:\r\n|\r|\n))\s*/g, '').trim().startsWith('('))
+	) {
+		const captured = /^((?:get|set)?(?:async\s*)?(?:(?:\/\*[^]*?\*\/|\/\/[^]*?(?:\r\n|\r|\n))\s*)*(?:function)?\s*(?:(?:\/\*[^]*?\*\/|\/\/[^]*?(?:\r\n|\r|\n))\s*)*\*?\s*(?:(?:\/\*[^]*?\*\/|\/\/[^]*?(?:\r\n|\r|\n))\s*)*)([^]*?)\s*(?:(?:\/\*[^]*?\*\/|\/\/[^]*?(?:\r\n|\r|\n))\s*)*\(/.exec(functionString);
```

Comment:
This regex is way too long and complicated.

Reply:
Do you have any suggestions on reducing the size? It is the only way to get the program working.

Reply:
Like I commented earlier, strip the comments in an initial pass and then to the replacement.

Reply:
The current `RegExp`, preserves comments and whitespace. Stripping the comments would not preserve it.

Reply:
https://github.com/sindresorhus/rename-fn/pull/3#issuecomment-760578829